File: java/functions/FileOperations.java
import java.io.*;
import java.nio.file.*;

public class FileOperations {
    public static void main(String[] args) {
        String testFile = "../tmp/test_output.txt";

        try {
            String data = "Hello from Java!\nThis is a test file.\n";
            Files.write(Paths.get(testFile), data.getBytes());
            System.out.println("Written to " + testFile);

            String content = new String(Files.readAllBytes(Paths.get(testFile)));
            System.out.println("Read content:\n" + content);

            if (Files.exists(Paths.get(testFile))) {
                System.out.println("File exists: " + testFile);
                System.out.println("File size: " + Files.size(Paths.get(testFile)) + " bytes");
            }

            Files.write(Paths.get(testFile), "Appended line\n".getBytes(), StandardOpenOption.APPEND);
            System.out.println("Appended to file");

        } catch (IOException e) {
            System.err.println("Error: " + e.getMessage());
        }
    }
}

File: rust/functions/file_operations.rs
use std::fs;
use std::io::Write;

fn main() {
    let test_file = "../tmp/test_output.txt";

    let data = "Hello from Rust!\nThis is a test file.\n";
    fs::write(test_file, data).expect("Unable to write file");
    println!("Written to {}", test_file);

    let content = fs::read_to_string(test_file).expect("Unable to read file");
    println!("Read content:\n{}", content);

    if let Ok(metadata) = fs::metadata(test_file) {
        println!("File exists: {}", test_file);
        println!("File size: {} bytes", metadata.len());
    }

    let mut file = fs::OpenOptions::new()
        .append(true)
        .open(test_file)
        .expect("Unable to open file");

    file.write_all(b"Appended line\n").expect("Unable to append");
    println!("Appended to file");
}
